runtime/src/main/java/org/corfudb/runtime/view/AddressSpaceView.java

```diff
@@ -256,15 +262,28 @@ public void write(IToken token, Object data) throws OverwriteException {
                     .peek(e, address));
     }
 
+    /**
+     * Perform a single read with the default read options
+     */
+    public @NonNull ILogData read(long address) {
+        return read(address, defaultReadOptions);
+    }
+
     /**
      * Read the given object from an address and streams.
      *
      * @param address An address to read from.
+     * @param options Read options for this particular write (i.e. configure caching behavior)
      * @return A result, which be cached.
      */
     public @Nonnull
-    ILogData read(long address) {
-        if (!runtime.getParameters().isCacheDisabled()) {
+    ILogData read(long address, ReadOptions options) {
+        if (runtime.getParameters().isCacheDisabled() && options.isClientCacheable()) {
```

Comment:
Can we move this sanity check into `private boolean cacheReadRequest(ReadOptions options) {}` ?

Reply:
done.